### DISCUSSION TITLE
About merging two different path objects

### DISCUSSION THREAD

**Discussion by ekilcioglu** [NVlabs/sionna#335] [Q&A] [answered]
Hello,

I try to manually model reconfigurable intelligent surfaces (RISs) inside an indoor scenario using Sionna RT (I know Sionna does not have support for RIS at the moment). I will have 1 transmitter, 1 receiver and 1 RIS in the scenario. Regarding this issue, I have two questions:

1) Can I do the following?: First, define RIS as a receiver and compute the paths between the transmitter and RIS with scene.compute_paths. Then, define RIS as a transmitter and compute the paths between the RIS and the receiver. Then, can I merge these two different path objects and see the merged paths visually via preview() or render() in order to output RIS-aided communication paths between a transmitter and a receiver? Is there a way to merge these two different paths and visually see the output? I know there are **to_dict()** and **from_dict(data_dict)** methods to store the current path object properties and to set the defined path but what parameter should I change if the solution is related to modifying these methods?

2) This is independent from the RIS. Can we deploy our own pattern in the transmitter other than the predefined 4 patterns? If yes, where and how can we do it?

You can see the source code as well as the indoor scenario's Blender file in attached. Waiting for your help, thank you so much in advance!

[Uploading test_Sionna.zip…]()

**Comment by ekilcioglu** [comment]
[test_Sionna.zip](https://github.com/NVlabs/sionna/files/14295072/test_Sionna.zip)

**Comment by jhoydis** [comment, marked as answer]
Hi,

Concerning part 1), I do not think that this is a good idea. You could compute manually the additional paths due to the RIS and add them to the channel impulse response but I would not hack the ray visualisation.

Concerning 2), what do you mean by pattern? If you refer to the antenna pattern, you can simply define your own if you want. Have a look at the API of [``Antenna``](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.Antenna). You need to provide a ``pattern`` which can be any callable you want with the right input/output specification.

**Reply by ekilcioglu** [reply]
Thank you for your answer!

**Reply by sunskyday** [reply]
Hi,
scene.tx_array = AntennaArray(antenna=Antenna("dipole", "V"),
                             positions=tf.Variable([[-0.15,0,0], [0.15,0,0]]))

There are two TX antennas, and the two antennas can only work in one pattern and polarization mode. How to define the pattern and polarization of each antenna?

**Reply by merlinND** [reply]
@sunskyday Please refrain from posting the same question multiple times as this creates a lot of work for maintainers. Your question is being discussed in #631.
